### ISSUE TITLE
Adam optimizer results in low mAP

### ISSUE THREAD

**Issue by zhiqiangwan** [pierluigiferrari/ssd_keras#211]
I tried to use ssd300_triaining to train a SSD from scratch with VGG16 weights on VOC 07+12. When I used Adam optimizer, the training process is stable. However, the mAP on VOC 07 test is around 0.48. When I used SGD with momentum, the mAP is much higher and I can reproduce your result.

**Comment by chuong98** [comment]
@zhiqiangwan I have tried Adam, but got only up 0.4 after 120 epochs. Thanks for let us know that SGD works better. May I know if you have tuned any hyper parameters for SGD, e.g learning rate, momentum,..?

**Comment by zhiqiangwan** [comment]
@chuong98 I just used the default hyper parameters in this code.

**Comment by pierluigiferrari** [comment]
That's strange, you would think that Adam would perform equally well or better than SGD. I'll investigate this when I find the time. Something is definitely odd if Adam performs that much worse than SGD.

One piece of advice for others who try out Adam instead of SGD: If the mAP isn't somewhere at least in the high fifty percent range after 20k training steps, something is likely wrong, so don't waste your time completing the the full 120k training steps. Since the incremental training gains decrease exponentially with the number of iterations, the majority of the learning always happens in the early phase of the training.

**Comment by nassarofficial** [comment]
It's the same case on my own dataset, Adam is performing poorly...

**Comment by xuannianz** [comment]
Same to me.

**Comment by DonghoonPark12** [comment]
OMG. Is that reason of **Adam optimizer**?? I thought that I didn't try weight sampling in **trainval35k** weight tha code author provides. *>*!!
